File: android/app/src/main/java/noah/moohvie/models/WatchedMovie.kt
package noah.moohvie.models

import java.util.Date

data class WatchedMovie(
    val id: Int,
    val title: String,
    val posterPath: String?,
    val voteAverage: Double,
    val watchedDate: Date,
    val personalRating: Int = 0,
    val genreIDs: List<Int> = emptyList(),
    val pointsEarned: Int = 0,
    val releaseDate: String? = null,
) {
    val posterUrl: String?
        get() = posterPath?.let { "https://image.tmdb.org/t/p/w500$it" }

    val primaryGenreName: String
        get() = genreIDs.firstOrNull()?.let { GenreLookup.name(forId = it) } ?: "Non classé"
}

File: Moohvie/Models/watchedmovie.swift
import Foundation

struct WatchedMovie: Identifiable, Codable {
    let id: Int
    let title: String
    let posterPath: String?
    let voteAverage: Double
    let watchedDate: Date
    var personalRating: Int = 0
    var genreIDs: [Int] = []
    var pointsEarned: Int = 0
    var releaseDate: String? = nil

    var posterURL: URL? {
        guard let posterPath else { return nil }
        return URL(string: "https://image.tmdb.org/t/p/w500\(posterPath)")
    }

    var primaryGenreName: String {
        guard let firstID = genreIDs.first else { return "Non classé" }
        return GenreLookup.name(for: firstID)
    }
}
